app/Community/Controllers/UserSettingsController.php

```diff
@@ -109,6 +111,19 @@ public function updateProfile(UpdateProfileRequest $request): JsonResponse
         return response()->json(['success' => true]);
     }
 
+    public function updateLocale(UpdateLocaleRequest $request): JsonResponse
+    {
+        $data = UpdateLocaleData::fromRequest($request);
+
+        /** @var User $user */
+        $user = $request->user();
+
+        $user->locale = $data->locale;
```

Comment:
Should this validate the existence of the the lang/xxx.json file?

Reply:
Good idea. Added this validation rule and another test case in latest.